src/components/HexagonGrid.tsx

```diff
@@ -82,10 +87,30 @@ function HexagonGrid(props: HexagonGridProps) {
   };
 
   function testOnClick(hexagon: HexData) {
-    console.log("clicked! now", hexagon);
+    console.log("clicked! now", hexagon,"theme", brushSelection);
+    const newHexagon = hexagons.find(hex => hex.index === hexagon.index) || { index: hexagon.index } as HexData
+    if (brushSelection.icon !== undefined) {
+      newHexagon.bonus = {
+        ...newHexagon.bonus,
+        icon: brushSelection.icon
+      }
+    }
+    if (brushSelection.terrain !== undefined) {
+      newHexagon.terrain = brushSelection.terrain;
+    }
+    const foundIndex = hexagons.findIndex(hex => hex.index === hexagon.index);
+    if (foundIndex >= 0) {
+      hexagons[foundIndex] = {
+        ...hexagons[foundIndex],
```

Comment:
i just realized, this line isn't necessary, because we've already mutated the hexagon. `newHexagon` is not a good name because it is exactly the old hexagon we found, or a blank hexagon. We can do this instead

```
const hex = hexagons.find(...);
if (!hex) {
  const newHexagon = {
    terrain: brushSelection.terrain,
    bonus: brushSelection.icon ? {icon: brushSelection.icon} : undefined,
  };
  hexagons.push(newHexagon);
}
else {
  hexagon.terrain = brushSelection.terrain !== undefined 
    ? brushSelection.terrain 
    : hexagon.terrain;
  if (brushSelection.icon !== undefined) {
    hexagon.bonus = {
      ...hexagon.bonus,
      icon: brushSelection.icon
    };
  }
}
setHexagons([...hexagons]);
```